python/ml4ir/applications/ranking/tests/test_ml4ir_sanity.py

```diff
@@ -0,0 +1,258 @@
+        for i in range(len(X[xid])):
+            r['f' + str(i)] = X[xid][i]
+        r['clicked'] = label
+
+    clicked_docs = X[Y == 1]
+    nonclicked_docs = X[Y == 0]
+    consumed_clicked_docs = 0
+    consumed_nonclicked_docs = 0
+    rows = []
+    query_count = 0
+    document_count = 0
+    while consumed_clicked_docs < len(clicked_docs) and consumed_nonclicked_docs < len(nonclicked_docs):
+        qid = 'query'+str(query_count)
+        query_count += 1
+        r = {'query_id':qid}
+        docs_count = random.randint(2, 25)
+        r['record_id'] = 'record'+str(document_count)
+        r['fr'] =docs_count
+        add_features_to_record(clicked_docs, consumed_clicked_docs, r, 1)
+        consumed_clicked_docs += 1
+        document_count += 1
+        rows.append(r)
+        for i in range(docs_count-1):
+            if consumed_nonclicked_docs < len(nonclicked_docs):
+                r = {'query_id': qid}
+                r['fr'] = i+1
+                r['record_id'] = 'record' + str(document_count)
+                add_features_to_record(nonclicked_docs, consumed_nonclicked_docs, r, 0)
+                consumed_nonclicked_docs += 1
+                document_count += 1
+                rows.append(r)
+            else:
+                break
+    return pd.DataFrame(rows)
+
+
+def generate_data(args):
+    """
+    Generate data with the specified parameters
+    - n_classes: Number of classes in the datasets. Should be 2 (clicked, not clicked)
+    - n_samples: Number of samples to generate
+    - n_features: Number of features per sample
+    - n_redundant: The number of redundant features
+    - n_informative: The number of informative (Non-redundant) features
+    - n_clusters_per_class: The number of clusters per class.
+    - flip_y: The fraction of samples whose class is assigned randomly. Larger
+        values introduce noise in the labels and make the classification
+        task harder.
+    - class_sep: Larger values spread out the clusters/classes and make the classification task easier.
+    - seed: Random seed
+    Returns: the best seprabale data set by a Perceptorn.
+    """
+    separable = False
+    seed = args.seed
+    rng = check_random_state(args.seed)
+    best_accuracy = 0
+    best_accuracy_data = None
+    trials = 1000
+    while not separable and trials > 0:
+        samples = datasets.make_classification(n_classes=args.n_classes, n_samples=args.n_samples, n_features=args.n_features,
+                                               n_redundant=args.n_redundant, n_informative=args.n_informative,
+                                               n_clusters_per_class=args.n_clusters_per_class, flip_y=args.flip_y,
+                                               class_sep=args.class_sep, random_state=rng)
+        p = Perceptron(fit_intercept=True, max_iter=1000, verbose=0, random_state=seed, validation_fraction=0.0001)
+        p.fit(samples[0], samples[1])
+        acc = p.score(samples[0], samples[1])
```

Comment:
I checked and it's only classification accuracy.

Reply:
I assume you did (and I also know it because I like sklearn a lot). What I am trying to say is that for someone who does not know this price of information using 
```
sklearn.metrics.accurcy_score(.., ..)
``` 
is self-explanatory.

Reply:
I did not remember this because incidentally I have not used sklearn in a while.

Reply:
I think what also is a useful thing to incorporate (I include myself as well) is to write code like
```
...score(y_true=samples[0], y_pred=samples[1])
```